NorthstarDLL/util/printcommands.cpp

```diff
@@ -160,14 +200,93 @@ void ConCommand_findflags(const CCommand& arg)
 	delete[] upperFlag;
 }
 
+void ConCommand_list(const CCommand& arg)
+{
+	ConCommandBase* var;
+	CCVarIteratorInternal* itint = R2::g_pCVar->FactoryInternalIterator();
+	std::unordered_map<std::string, ConCommandBase*> unsortedConvars;
+	for (itint->SetFirst(); itint->IsValid(); itint->Next())
+	{
+		var = itint->Get();
+		if (!var->IsFlagSet(FCVAR_DEVELOPMENTONLY) && !var->IsFlagSet(FCVAR_HIDDEN))
+		{
+			unsortedConvars.insert({var->m_pszName, var});
+		}
+	}
+	delete itint;
+
+	std::vector<std::pair<std::string, ConCommandBase*>> sorted = ConvarSort(unsortedConvars);
+
+	for (auto& map : sorted)
+	{
+		PrintCommandHelpDialogue(map.second, map.second->m_pszName);
+	}
+}
+
+void ConCommand_differences(const CCommand& arg)
+{
+	ConCommandBase* var;
+	CCVarIteratorInternal* itint = R2::g_pCVar->FactoryInternalIterator();
+	std::unordered_map<std::string, ConCommandBase*> unsortedConvars;
+	for (itint->SetFirst(); itint->IsValid(); itint->Next())
+	{
+		var = itint->Get();
+		if (!var->IsFlagSet(FCVAR_DEVELOPMENTONLY) && !var->IsFlagSet(FCVAR_HIDDEN))
+		{
+			unsortedConvars.insert({var->m_pszName, var});
+		}
+	}
+	delete itint;
+
+	std::vector<std::pair<std::string, ConCommandBase*>> sorted = ConvarSort(unsortedConvars);
+
+	for (auto& map : sorted)
+	{
+		ConVar* cvar = R2::g_pCVar->FindVar(map.second->m_pszName);
+		if (cvar && !cvar->IsFlagSet(FCVAR_HIDDEN))
+		{
+			if (strcmp(cvar->GetString(), "FCVAR_NEVER_AS_STRING") != NULL)
+			{
+				if (strcmp(cvar->GetString(), cvar->m_pszDefaultValue) != NULL)
+				{
+					std::string formatted =
+						fmt::format("\"{}\" = \"{}\" ( def. \"{}\" )", cvar->GetBaseName(), cvar->GetString(), cvar->m_pszDefaultValue);
+					if (cvar->m_bHasMin)
+					{
+						formatted.append(fmt::format(" min. {}", cvar->m_fMinVal));
+					}
+					if (cvar->m_bHasMax)
+					{
+						formatted.append(fmt::format(" max. {}", cvar->m_fMaxVal));
+					}
+					formatted.append(fmt::format(" - {}", cvar->GetHelpText()));
+					spdlog::info(formatted);
+				}
+			}
+		}
```

Comment:
Would prefer guard clauses here instead of nesting

Reply:
Resolved in fa9b19e5b7b3b86610b096debca674f5155340be